### PR TITLE
MP-903 Fork Terra faucet

### PR DESCRIPTION
**Pull request by donovansolms**
[JIRA Issue](https://delphilabs.atlassian.net/browse/MP-903)

This PR implements the changes required for the faucet to work on the Mars Ares-1 testnet.
PagerDuty sections have been commented out in case we'd like to use it in future.

This PR does not cover the styling and design of the faucet.

### CODE REVIEW

File: faucet.go [hunk 1]
```diff
@@ -494,22 +496,22 @@ func signAndBroadcast(txBuilder client.TxBuilder, isDetectMismatch bool) string
...
-			Type: "service",
-		},
-		Body: &pagerduty.APIDetails{
-			Details: body,
-		},
-	}
-
-	return client.CreateIncidentWithContext(context.Background(), pagerdutyConfig.user, input)
-}
+// func createIncident(body string) (*pagerduty.Incident, error) {
```
Comment (donovansolms) [inline faucet.go:499]: We can enable this again in future should we need it

File: frontend/src/config.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 export const networks = [
   {
-    chainId: 'pisco-1',
-    faucetUrl: 'https://pisco-faucet.terra.dev/claim',
+    chainId: 'ares-1',
+    faucetUrl: 'https://faucet.marsprotocol.io/claim',
```
Comment (donovansolms) [inline frontend/src/config.js:4]: This endpoint isn't live yet, but will be soon.

Comment (larry0x) [reply frontend/src/config.js:4]: can we make the URL to be just https://faucet.marsprotocol.io without the "claim" part?

Comment (donovansolms) [reply frontend/src/config.js:4]: This is just the FE -> BE API call, users will access the faucet page via [https://faucet.marsprotocol.io](https://faucet.marsprotocol.io)

File: faucet.go [hunk 3]
```diff
@@ -30,10 +29,8 @@ import (
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 
 	"github.com/cosmos/cosmos-sdk/x/auth/signing"
-	"github.com/terra-money/core/v2/app"
-	"github.com/terra-money/core/v2/app/params"
+	"github.com/mars-protocol/faucet/params"
```
Comment (larry0x) [inline faucet.go:32]: you can use this dependency: 

```
import marsapp "github.com/mars-protocol/hub/app"
```

i have defined `marsapp.MakeEncodingConfig` here: https://github.com/mars-protocol/hub/blob/main/app/encoding.go

Comment (donovansolms) [reply faucet.go:32]: Ah great, I was looking for a params module in hub, but didn't find it. Will update

File: faucet.go [hunk 4]
```diff
@@ -69,17 +66,22 @@ var pagerdutyConfig PagerdutyConfig
...
+	accountPubKeyPrefix    = "marspub"
+	validatorAddressPrefix = "marsvaloper"
+	validatorPubKeyPrefix  = "marsvaloperpub"
+	consNodeAddressPrefix  = "marsvalcons"
+	consNodePubKeyPrefix   = "marsvalconspub"
+
+	CoinType    = 118
+	CoinPurpose = 44
+	// BondDenom staking denom
+	BondDenom = "umars"
```
Comment (larry0x) [inline faucet.go:80]: i think `BondDenom` should be included in `marsapp` package as well so that you don't need to define it here

i'll do this shortly

**Review by larry0x — APPROVED** [review]

File: faucet.go (hunk 4 above)
Comment (larry0x) [reply faucet.go:80]: @donovansolms added: https://github.com/mars-protocol/hub/commit/5eb26302567d4d64a1f6b3f9c6282bb9bbcac2da

File: faucet.go [hunk 5]
```diff
@@ -168,7 +167,7 @@ type BalanceResponse struct {
 }
 
 func getBalance(address string) (amount int64) {
-	url := fmt.Sprintf("%v/cosmos/bank/v1beta1/balances/%v/by_denom?denom=uluna", lcdURL, address)
+	url := fmt.Sprintf("%v/cosmos/bank/v1beta1/balances/%v/by_denom?denom=umars", lcdURL, address)
```
Comment (larry0x) [inline faucet.go:170]: use `marsapp.BondDenom` here?

File: faucet.go (hunk 4 above)
Comment (donovansolms) [reply faucet.go]: Updated. Address prefix is now also takes from Hub instead

File: faucet.go (hunk 5 above)
Comment (larry0x) [inline faucet.go:170]:
```suggestion
	url := fmt.Sprintf("%v/cosmos/bank/v1beta1/balances/%v/by_denom?denom=%s", lcdURL, address, marsapp.BondDenom)
```

File: frontend/src/components/Home.js [hunk 6]
```diff
@@ -205,7 +205,7 @@ class HomeComponent extends React.Component {
           <article>
             There's two ways to get one. The first is by using Station, the
             crypto wallet for Terra. If you know command-line-fu, you can also
-            generate an address with the Terra SDK.
+            generate an address with the Mars SDK.
```
Comment (markonmars) [inline frontend/src/components/Home.js:208]: There is a Mars SDK? 😮

**Review by markonmars — APPROVED** [review]

File: frontend/src/components/Home.js (hunk 6 above)
Comment (larry0x) [reply frontend/src/components/Home.js:208]: > There is a Mars SDK? 😮

there isn't... @donovansolms can you change "Mars SDK" to "Mars daemon application (marsd)" and link to the mars hub repository?

Comment (donovansolms) [reply frontend/src/components/Home.js]: Done. The frontend part will be overhauled when the design for it is ready, apologies for the few lingering Terra references there

**Pull request merged**